src/wp-includes/option.php

```diff
@@ -1508,6 +956,24 @@ function get_transient( $transient ) {
  */
 function set_transient( $transient, $value, $expiration = 0 ) {
 
+	// Transient name should not be empty string, 0, or false.
+	if ( empty( $transient ) ) {
+		return false;
+	}
+
+	// Transient name should be a string or number; bool false is rejected in previous condition.
+	if ( true === $transient || ! is_scalar( $transient ) ) {
+		return false;
+	}
+
+	// Cast transient name as string.
+	$transient = (string) $transient;
+
+	// Check that Transient name is less than or equal to 172 characters.
+	if ( 172 <= strlen( $transient ) ) {
+		return false;
```

Comment:
This could add a `_doing_it_wrong()` to warn about why setting the transient failed.